xarray/core/common.py

```diff
@@ -211,14 +211,27 @@ def _ipython_key_completions_(self):
         return list(set(item_lists))
 
 
-def get_squeeze_dims(xarray_obj, dim):
+def get_squeeze_dims(xarray_obj, dim, axis=None):
     """Get a list of dimensions to squeeze out.
     """
-    if dim is None:
+    if not dim is None and not axis is None:
+        raise ValueError('cannot use both parameters `axis` and `dim`')
+
+    if dim is None and axis is None:
         dim = [d for d, s in xarray_obj.sizes.items() if s == 1]
     else:
         if isinstance(dim, basestring):
             dim = [dim]
+        if isinstance(axis, int):
+            alldims = list(xarray_obj.sizes.keys())
+            dim = [alldims[axis]]
+        if isinstance(axis, tuple):
+            for a in axis:
+                if not isinstance(a, int):
+                    raise ValueError(
+                        'parameter `axis` must be int or tuple of int.')
+            alldims = list(xarray_obj.sizes.keys())
+            dim = [alldims[a] for a in axis]
```

Comment:
I think this can be simpler
`dim = [xarray_obj.dims[a] for a in axis]`

Reply:
`xarray_obj` can be either a Dataset or a DataArray, so I think this is indeed necessary.

Reply:
You are right. Let's keep this line.